src/Middleware/RequestDecompression/test/RequestDecompressionMiddlewareTests.cs

```diff
@@ -84,8 +92,32 @@ public async Task Request_ContentEncodingBrotli_Decompressed()
     }
 
     [Fact]
-    public async Task Request_ContentEncodingDeflate_Decompressed()
+    public async Task Request_ContentEncodingDeflate_ZlibCompressed_Decompressed()
+    {
+        // This tests the "correct" version of deflate usage.
+        // As described in RFC 2616, the deflate content-coding is the "zlib" format (RFC 1950)
+        // in combination with the "deflate" compression algorithm (RFC 1951).
+
+        // Arrange
+        var contentEncoding = "deflate";
+        var uncompressedBytes = GetUncompressedContent();
+        var compressedBytes = await GetZlibCompressedContent(uncompressedBytes);
+
+        // Act
+        var (logMessages, decompressedBytes) = await InvokeMiddleware(compressedBytes, new[] { contentEncoding });
+
+        // Assert
+        AssertDecompressedWithLog(logMessages, contentEncoding.ToLowerInvariant());
+        Assert.Equal(uncompressedBytes, decompressedBytes);
+    }
+
+    [Fact]
```

Comment:
is there a test for the case where the stream cannot be read (even a byte)? it looks like before and after this change that would end up using Deflate Stream (because the read will return -1 and (firstByte & 0xF) will not  be 8). But maybe there should be a test that verifies it doesn't throw in this path,